src/highlight/HighlightPromoter.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import classNames from 'classnames';
+import { getSelectionRange, isInRect } from './highlightUtil';
+import { MOUSE_PRIMARY } from '../constants';
+import './HighlightPromoter.scss';
+
+type Props = {
+    className?: string;
+    onSelectionChange?: (range: Range | null) => void;
+    pageEl: HTMLElement;
+};
+
+export default function HighlightPromoter({ className, onSelectionChange, pageEl }: Props): JSX.Element {
+    const [isSelecting, setIsSelecting] = React.useState<boolean>(false);
+
+    const handleMouseDown = ({ buttons }: MouseEvent): void => {
+        if (!onSelectionChange || buttons !== MOUSE_PRIMARY) {
+            return;
+        }
+
+        setIsSelecting(true);
+    };
+
+    const handleMouseUp = ({ clientX, clientY }: MouseEvent): void => {
+        if (!onSelectionChange || !isSelecting) {
+            return;
+        }
+
+        setIsSelecting(false);
+
+        // Multi-page highlight is not supported
+        if (!isInRect(clientX, clientY, pageEl.getBoundingClientRect())) {
+            return;
+        }
+
+        setTimeout(() => {
+            onSelectionChange(getSelectionRange());
```

Comment:
Instead of calling `onSelectionChange`, would this component get the bounding rect and use that to render the promoter popover?

Reply:
Will render popover in the next PR